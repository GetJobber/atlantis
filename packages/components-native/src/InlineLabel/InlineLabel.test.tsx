import React from "react";
import { render } from "@testing-library/react-native";
import { InlineLabel } from ".";

it("renders inline label with no additional props", () => {
  const inlineLabel = render(<InlineLabel>Draft</InlineLabel>).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});
it("renders inline label with size default", () => {
  const inlineLabel = render(
    <InlineLabel size="default">Draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with size large", () => {
  const inlineLabel = render(
    <InlineLabel size="large">Draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with size larger", () => {
  const inlineLabel = render(
    <InlineLabel size="larger">Draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with color red", () => {
  const inlineLabel = render(
    <InlineLabel color="red">Draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with color teal", () => {
  const inlineLabel = render(
    <InlineLabel color="teal">Draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with color and size", () => {
  const inlineLabel = render(
    <InlineLabel color="teal" size="large">
      Draft
    </InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});

it("renders inline label with capitalize transform", () => {
  const inlineLabel = render(
    <InlineLabel transform="capitalize">draft</InlineLabel>,
  ).toJSON();
  expect(inlineLabel).toMatchSnapshot();
});
