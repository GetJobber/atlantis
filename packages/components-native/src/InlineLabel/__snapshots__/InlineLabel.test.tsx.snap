// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inline label with capitalize transform 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(195, 12%, 94%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    Draft
  </Text>
</View>
`;

exports[`renders inline label with color and size 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 14,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(190, 63%, 92%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(190, 78%, 25%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 16,
          "lineHeight": 20,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with color red 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(7, 63%, 95%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(6, 100%, 24%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with color teal 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(190, 63%, 92%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(190, 78%, 25%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with no additional props 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(195, 12%, 94%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with size default 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 8,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(195, 12%, 94%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with size large 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 14,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "hsl(195, 12%, 94%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 16,
          "lineHeight": 20,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;

exports[`renders inline label with size larger 1`] = `
<View
  style={
    [
      {
        "borderRadius": 16,
      },
      {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
      {
        "backgroundColor": "hsl(195, 12%, 94%)",
      },
    ]
  }
>
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-regular",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 18,
          "lineHeight": 22,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    DRAFT
  </Text>
</View>
`;
